scripts/oe-env-prepare.ps1

```diff
@@ -0,0 +1,446 @@
+    $installDir = Join-Path $PACKAGES_DIRECTORY "intel_psw_install"
+    Install-ZipTool -ZipPath $PACKAGES["psw"]["local_file"] `
+                    -InstallDirectory $installDir
+
+    $pswInstaller = Join-Path $installDir "Intel SGX PSW for Windows v2.3.100.49777\PSW_EXE_RS2_and_before\Intel(R)_SGX_Windows_x64_PSW_2.3.100.49777.exe"
+    $p = Start-Process -Wait -NoNewWindow -PassThru -FilePath $pswInstaller `
+                       -ArgumentList @("--extract-folder", "$installDir", "--x")
+    if($p.ExitCode -ne 0) {
+        Throw "Failed to extract the Intel SGX PSW bundle: $pswInstaller"
+    }
+    $p = Start-Process -Wait -NoNewWindow -PassThru -FilePath "$installDir\install.exe" `
+                       -ArgumentList @("install", "--eula=accept", "--output=$installDir\intel_install.log", "--components=all")
+    if($p.ExitCode -ne 0) {
+        Throw "Failed to install the Intel SGX PSW software"
+    }
+}
+
+
+function Install-VisualStudio {
+    $installerArguments = @(
+        "-q", "--wait", "--norestart", "--nocache",
+        "--add Microsoft.VisualStudio.Workload.MSBuildTools",
+        "--add Microsoft.VisualStudio.Workload.VCTools",
+        "--add Microsoft.VisualStudio.Component.VC.Tools.x86.x64",
+        "--add Microsoft.VisualStudio.Component.VC.140",
+        "--add Microsoft.VisualStudio.Component.Windows10SDK.16299.Desktop",
+        "--add Microsoft.VisualStudio.Component.Windows81SDK",
+        "--add Microsoft.VisualStudio.Component.VC.ATL"
+        "--add Microsoft.VisualStudio.Component.Windows10SDK.17134"
+        "--includeRecommended"
+    )
+    # VisualStudio install sometimes is throwing errors on first try.
+    Start-ExecuteWithRetry -ScriptBlock {
+       Install-Tool -InstallerPath $PACKAGES["vs_buildtools"]["local_file"] `
+                    -ArgumentList $installerArguments
+    } -RetryMessage "Failed to install Visual Studio. Retrying"
+
+    [Environment]::SetEnvironmentVariable("VS150COMNTOOLS", "${env:ProgramFiles(x86)}\Microsoft Visual Studio\2017\BuildTools\Common7\Tools", "Machine")
+}
+
+function Install-Cmake {
+    $installDir = Join-Path $env:ProgramFiles "CMake"
+
+    Install-Tool -InstallerPath $PACKAGES["cmake"]["local_file"] `
+                 -InstallDirectory $installDir `
+                 -ArgumentList @("/quiet", "/passive") `
+                 -EnvironmentPath @("$installDir\bin")
+
+}
+
+function Install-Ocaml {
+    $installDir = Join-Path $env:ProgramFiles "OCaml"
+    $tmpDir = Join-Path $PACKAGES_DIRECTORY "ocpwin64"
+    Install-ZipTool -ZipPath $PACKAGES["ocaml"]["local_file"] `
+                    -InstallDirectory $tmpDir `
+                    -EnvironmentPath @("$installDir\bin")
+    $extractDir = Get-ChildItem $tmpDir | Select-Object -Property Name
+    New-Directory -Path $installDir -RemoveExisting
+    Copy-Item  "$tmpDir\$($extractDir.Name)\*" -Destination $installDir -Recurse -Force
+    Push-Location "$installDir\bin"
+    & ocpwin -in
```

Comment:
I don't believe this is needed.

Looking at the [documentation](https://www.typerex.org/ocpwin-manual-install.html), step 2, this seems to register OCaml in the Windows registry and update the PATH.

However, the Ansible logic didn't execute this and we already update the PATH via `Install-ZipTool` at the beginning of the function. Please maintain the same Ansible logic that we have:
https://github.com/microsoft/openenclave/blob/a982b46cf440def8fb66e94f2622a4f81e2b350b/scripts/ansible/roles/windows/openenclave/tasks/environment-setup.yml#L45-L55